src/server/src/main/java/io/cassandrareaper/service/RepairManager.java

```diff
@@ -93,77 +93,82 @@ public void initializeThreadPool(
-      for (RepairRun repairRun : running) {
+      Collection<RepairRun> runningRepairRuns = context.storage.getRepairRunsWithState(RepairRun.RunState.RUNNING);
+      Collection<RepairRun> pausedRepairRuns = context.storage.getRepairRunsWithState(RepairRun.RunState.PAUSED);
+      abortAllRunningSegmentsWithNoLeader(runningRepairRuns);
+      abortAllRunningSegmentsInKnownPausedRepairRuns(pausedRepairRuns);
+      resumeUnkownRunningRepairRuns(runningRepairRuns);
+      resumeUnknownPausedRepairRuns(pausedRepairRuns);
+    } catch (RuntimeException e) {
+      throw new ReaperException(e);
+    }
+  }
 
-        Collection<RepairSegment> runningSegments =
-            context.storage.getSegmentsWithState(repairRun.getId(), RepairSegment.State.RUNNING);
+  private void abortAllRunningSegmentsWithNoLeader(Collection<RepairRun> runningRepairRuns) throws ReaperException {
+    runningRepairRuns
+        .forEach((repairRun) -> {
+          Collection<RepairSegment> runningSegments
+              = context.storage.getSegmentsWithState(repairRun.getId(), RepairSegment.State.RUNNING);
 
-        abortSegmentsWithNoLeader(repairRun, runningSegments);
+          abortSegmentsWithNoLeader(repairRun, runningSegments);
+        });
+  }
 
-        if (!repairRunners.containsKey(repairRun.getId())) {
-          LOG.info("Restarting run id {} that has no runner", repairRun.getId());
-          startRepairRun(repairRun);
-        }
+  private void resumeUnkownRunningRepairRuns(Collection<RepairRun> runningRepairRuns) throws ReaperException {
+    for (RepairRun repairRun : runningRepairRuns) {
+      if (!repairRunners.containsKey(repairRun.getId())) {
+        LOG.info("Restarting run id {} that has no runner", repairRun.getId());
+        // it may be that this repair is already "running" actively on other reaper instances
+        //  nonetheless we need to make it actively running on this reaper instance as well
+        //   so to help in running the queued segments
+        startRepairRun(repairRun);
       }
+    }
+  }
 
-      Collection<RepairRun> paused =
-          context.storage.getRepairRunsWithState(RepairRun.RunState.PAUSED);
-      for (RepairRun pausedRepairRun : paused) {
-        if (repairRunners.containsKey(pausedRepairRun.getId())) {
+  private void abortAllRunningSegmentsInKnownPausedRepairRuns(Collection<RepairRun> pausedRepairRuns) {
+    pausedRepairRuns
+        .stream()
+        .filter((pausedRepairRun) -> repairRunners.containsKey(pausedRepairRun.getId()))
+        .forEach((pausedRepairRun) -> {
           // Abort all running segments for paused repair runs
-          Collection<RepairSegment> runningSegments =
-              context.storage.getSegmentsWithState(
-                  pausedRepairRun.getId(), RepairSegment.State.RUNNING);
+          Collection<RepairSegment> runningSegments
+              = context.storage.getSegmentsWithState(pausedRepairRun.getId(), RepairSegment.State.RUNNING);
```

Comment:
Yup :-)

I separated this change out as part of the first commit, which has the description:
> break down the RepairManager.resumeRunningRepairRuns() method for better readability

the four methods are all new, i was kinda hoping that the diff would have been a bit more higher level (one method replaced with five), rather than presented line-by-line.